src/compiler/binder.ts

```diff
@@ -2419,14 +2420,20 @@ function createBinder(): (file: SourceFile, options: CompilerOptions) => void {
             !(node.flags & NodeFlags.JSDoc) &&
             !isIdentifierName(node)) {
 
+
+            const originalKeywordKind = stringToToken(node.escapedText as string);
+            if (originalKeywordKind === undefined) {
+                return;
+            }

```

Comment:
Is this ever possible? Could it be an assert?

Reply:
This is just saying that if your identifier has no overlap with any keywords, you don't have to try doing error reporting. I could technically avoid the lookup entirely unless some of the other conditions succeed.

Reply:
Sure, I was just thinking about it in relation to the code below which previously just assumed the value would be defined. Or maybe, it worked because of undefined coercion?

Reply:
Yeah, comparisons with `undefined`ish things are not my favorite - I know @andrewbranch kind of likes them though. 😄